### PR TITLE
displaylink: 5.3.1 -> 5.4.0 (+evdi)

### PR DESCRIPTION
**Pull request by eyJhb**
###### Motivation for this change
New version of DisplayLink.

They have NOT tagged the evdi version that DisplayLink requires, hence the unstable tag.

I am not sure how to name the PR, once I am updating two packages in a single PR.
I am doing both in this, as they heavily depend on each other.

Tested on `652749c4ca7f2d4aa169fd4c24feb9ce8dc4fca9`, worked fine without a hitch.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eyJhb** [comment]
@nshalman ready for testing another round of DisplayLink? :)

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119664` at dc974c47 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.evdi</li>
    <li>linuxPackages_4_4.evdi</li>
    <li>linuxPackages_hardkernel_4_14.evdi</li>
    <li>linuxPackages_hardkernel_latest.evdi</li>
    <li>linuxPackages_latest-libre.evdi</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/73511c1f3551bee04589d6a79eb1b6b9">linuxPackages_4_14.evdi</a></li>
    <li><a href="https://gist.github.com/40b2bd31e659f745bffb52492b6d4a10">linuxPackages_4_9.evdi</a></li>
  </ul>
</details>
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>displaylink</li>
  </ul>
</details>
<details>
  <summary>14 packages built successfully:</summary>
  <ul>
    <li>linuxPackages.evdi (linuxPackages_5_10.evdi)</li>
    <li>linuxPackages_4_19.evdi</li>
    <li>linuxPackages_5_11.evdi (linuxPackages_latest.evdi)</li>
    <li>linuxPackages_5_4.evdi</li>
    <li>linuxPackages_hardened.evdi</li>
    <li>linuxPackages_latest_hardened.evdi</li>
    <li>linuxPackages_latest_xen_dom0.evdi</li>
    <li>linuxPackages_latest_xen_dom0_hardened.evdi</li>
    <li>linuxPackages_lqx.evdi</li>
    <li>linuxPackages_testing_bcachefs.evdi</li>
    <li>linuxPackages_xanmod.evdi</li>
    <li>linuxPackages_xen_dom0.evdi</li>
    <li>linuxPackages_xen_dom0_hardened.evdi</li>
    <li>linuxPackages_zen.evdi</li>
  </ul>
</details>
<details>
  <summary>7 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `systemd`.
Near pkgs/os-specific/linux/displaylink/default.nix:7:3:
```
  |
7 | , systemd
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/evdi/default.nix:34:5:
```
   |
34 |     license = with licenses; [ lgpl21 gpl2 ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/evdi/default.nix:34:5:
```
   |
34 |     license = with licenses; [ lgpl21 gpl2 ];
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/os-specific/linux/evdi/default.nix:1:33:
```
  |
1 | { lib, stdenv, fetchFromGitHub, fetchpatch, kernel, libdrm }:
  |                                 ^
```
</li>
    <li>warning: unused-argument

Unused argument: `substituteAll`.
Near pkgs/os-specific/linux/displaylink/default.nix:10:3:
```
   |
10 | , substituteAll
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/os-specific/linux/displaylink/default.nix:51:3:
```
   |
51 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/os-specific/linux/evdi/default.nix:25:3:
```
   |
25 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Review by jonringer — COMMENTED** [review]
looks like evdi has dropped support for `4.14` and `4.9`, please update the `broken` attribute

```
  broken = versionOlder kernel.version "4.19" || stdenv.isAarch64;
```
```
3 packages failed to build:
displaylink linuxPackages_4_14.evdi linuxPackages_4_9.evdi
```

**Comment by eyJhb** [comment]
> looks like evdi has dropped support for `4.14` and `4.9`, please update the `broken` attribute
> 
> ```
>   broken = versionOlder kernel.version "4.19" || stdenv.isAarch64;
> ```
> 
> ```
> 3 packages failed to build:
> displaylink linuxPackages_4_14.evdi linuxPackages_4_9.evdi
> ```

Fixed in newest commit :)

**Comment by nshalman** [comment]
> @nshalman ready for testing another round of DisplayLink? :)

I don't currently have a system set up for testing this. :(
Does either of these commits depend on the other? Should they be upgraded in lockstep together and thus in a single commit?

**Comment by eyJhb** [comment]
> > @nshalman ready for testing another round of DisplayLink? :)
> 
> I don't currently have a system set up for testing this. :(
> Does either of these commits depend on the other? Should they be upgraded in lockstep together and thus in a single commit?

They depend on each other, but I think we have to keep them separate. At least that is what I got from `#nixos-dev` :)

File: pkgs/os-specific/linux/displaylink/default.nix [hunk 1]
```diff
@@ -20,17 +20,17 @@ let
...
     name = "displaylink.zip";
-    sha256 = "1c1kbjgpb71f73qnyl44rvwi6l4ivddq789rwvvh0ahw2jm324hy";
+    sha256 = "1m2l3bnlfwfp94w7khr05npsbysg9mcyi7hi85n78xkd0xdcxml8";
     message = ''
       In order to install the DisplayLink drivers, you must first
       comply with DisplayLink's EULA and download the binaries and
       sources from here:
 
-      https://www.displaylink.com/downloads/file?id=1576
+      https://www.synaptics.com/node/3751
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/displaylink/default.nix:33]:
```suggestion
      https://www.synaptics.com/products/displaylink-graphics/downloads/ubuntu-5.4
```

Comment (eyJhb) [reply pkgs/os-specific/linux/displaylink/default.nix:33]: https://logs.nix.samueldr.com/nixos-dev/2021-04-22#1619082865-1619082653;

**Review by nshalman — APPROVED** [review]
Thanks for keeping this up to date!

**Pull request merged**